ampl-executor/src/main/java/com/powsybl/ampl/executor/AmplModelExecutionHandler.java

```diff
@@ -0,0 +1,153 @@
+ * {@link AmplExporter#export}.
+ * <p>
+ * Then it runs the ampl model, and {@link AmplReadableElement#readElement} is used to
+ * apply modifications on the network.
+ * <p>
+ * The majority of the configuration is made by the {@link IAmplModel}
+ * interface.
+ * @author Nicolas Pierre <nicolas.pierre@artelys.com>
+ */
+public class AmplModelExecutionHandler extends AbstractExecutionHandler<AmplResults> {
+
+    private static final String AMPL_BINARY = "ampl";
+    private static final String COMMAND_ID = "AMPL_runner";
+    private final IAmplParameters parameters;
+    private final IAmplModel model;
+    private final Network network;
+    private final String networkVariant;
+    private final AmplConfig config;
+    private final StringToIntMapper<AmplSubset> mapper;
+
+    public AmplModelExecutionHandler(IAmplModel model, Network network, String networkVariant, AmplConfig config,
+                                     IAmplParameters parameters) {
+        this.model = model;
+        this.network = network;
+        this.networkVariant = networkVariant;
+        this.config = config;
+        this.parameters = parameters;
+        this.mapper = AmplUtil.createMapper(this.network);
+    }
+
+    /**
+     * This method will write ampl model files (.run, .dat and .mod)
+     *
+     * @param workingDir the directory where to write the files
+     * @throws IOException rethrow {@link Files#copy(InputStream, Path, CopyOption...)}
+     */
+    private void exportAmplModel(Path workingDir) throws IOException {
+        for (Pair<String, InputStream> fileAndStream : model.getModelAsStream()) {
+            try (InputStream modelStream = fileAndStream.getRight()) {
+                Files.copy(modelStream, workingDir.resolve(fileAndStream.getLeft()),
+                        StandardCopyOption.REPLACE_EXISTING);
+            }
+        }
+    }
+
+    /**
+     * This method will write parameters files.
+     *
+     * @param workingDir the directory where to write the files
+     * @throws IOException rethrow {@link Files#copy(InputStream, Path, CopyOption...)}
+     */
+    private void exportAmplParameters(Path workingDir) throws IOException {
+        for (IAmplInputFile param : parameters.getInputParameters()) {
+            try (InputStream paramStream = param.getParameterFileAsStream(this.mapper)) {
+                Files.copy(paramStream, workingDir.resolve(param.getFileName()), StandardCopyOption.REPLACE_EXISTING);
+            }
+        }
+    }
+
+    private void exportNetworkAsAmpl(Path workingDir) throws IOException {
+        DataSource networkExportPath = new FileDataSource(workingDir, this.model.getNetworkDataPrefix());
+        // AmplExporter throws UncheckedIOException
+        // but we are in a context where IOException are handled
+        // thus we rethrow a simple IOException
+        try {
+            new AmplExporter().export(network, null, networkExportPath);
+        } catch (UncheckedIOException rethrow) {
```

Comment:
We always try to wrap uncheck exceptions like  IOException as soon as possible to avoid to bother with it after. So here and more generally in this interface, you should do the invert and not re-wrap it into a check exception.